use-shopping-cart/core/stripe-middleware.js

```diff
@@ -0,0 +1,31 @@
+import { loadStripe } from '@stripe/stripe-js'
+import { getCheckoutData } from '../utilities/old-utils'
+
+export const handleStripe = (store) => (next) => async (action) => {
+  if (action.type === 'cart/redirectToCheckout') {
+    try {
+      const stripePublicKey = store.getState().stripe
+
+      const result = await loadStripe(stripePublicKey)
+      const cart = store.getState()
+
+      if (action.payload?.sessionId) {
+        return result.redirectToCheckout({
+          sessionId: action.payload.sessionId
+        })
+      }
+
+      if (store.getState().mode === 'client-only') {
+        const checkoutData = getCheckoutData(cart)
+        result.redirectToCheckout(checkoutData)
+      }
+    } catch (error) {
+      console.log('error', error)
+    }
```

Comment:
I think it would make more sense to wrap the `await loadStripe(stripePublicKey)` with the try-catch statement since it is the only async part. This way we could accurately provide an error log and then return that error so that the end developer can use it to show an accurate error state. Something like:

```js
let stripe
try {
  stripe = await loadStripe(stripePublicKey)
} catch (error) {
  console.error('Unable to load Stripe.js with the provided public key:', stripePublicKey, action, error)
  return error
}
```

The usage for when an error occurs would look something like:

```js
function Checkout() {
  const [status, setStatus] = useState('idle')
  const { redirectToCheckout } = useShoppingCart();
  async function checkout() {
    // Change status to render checking out indicator
    setStatus('checking-out')

    try {
      const sessionId = await // get id somehow
    } catch {
      // Change status to render failure to reach server for session id
      setStatus('failed-to-get-session-id')
    }

    const result = await redirectToCheckout({ sessionId })
    if (result instanceof Error) {
      // Change status to render failure to checkout with Stripe
      setStatus('failed-to-check-out')
    }
  }

  return /* ... */
}
```

Obviously, the end developer would likely want to handle the individual types of errors by using `instanceof SpecificErrorClass`.